typescript/src/marcos/StringCalculator.ts

```diff
@@ -2,6 +2,27 @@ export class StringCalculator {
   constructor() {}
 
   add(input) {
-    return null;
+    let response: number = 0;
+    let exceptionMessage: string = "negatives not allowed: ";
+    let negatives: number[] = [];
+    let demiliter: string = "";
+
+    demiliter = input.match(/\/\/(.*)\n/);
+    demiliter = demiliter ? demiliter[1] : null;
+
+    input = input ? input.split(demiliter ?? /[\n,]/) : [];
+
+    input = input.filter((n) => n <= 1000);
+    negatives = input.filter((n) => n < 0);
+
+    if (negatives.length) {
+      return exceptionMessage + negatives.join(",");
+    }
+
+    input.map((n) => {
```

Comment:
Tampoco tiene que ver con el kata, pero te podría ser útil. La semántica de `map` es para "mapear" una colección, es decir, transformar cada uno de los items.

En este caso, `forEach` funcionaría igual y tiene la semántica correcta.

```suggestion
    input.forEach((n) => {
```